File: doc-examples/jdbc-example-java/src/test/java/example/UserRepositorySpec.java
package example;

import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
class UserRepositorySpec {
    @Inject UserRepository userRepository;

    @Test
    void testSoftDelete() {
        final User joe = new User("Joe");
        final User fred = new User("Fred");
        final User bob = new User("Bob");
        userRepository.saveAll(Arrays.asList(
                fred,
                bob,
                joe
        ));

        userRepository.deleteById(joe.getId());

        assertEquals(2, userRepository.count());
        assertTrue(userRepository.existsById(fred.getId()));
        assertFalse(userRepository.existsById(joe.getId()));

        final List<User> disabled = userRepository.findDisabled();
        assertEquals(1, disabled.size());
        assertEquals("Joe", disabled.iterator().next().getName());
    }
}

File: doc-examples/jdbc-example-python/src/test/python/example/UserRepositorySpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from example.User import User
from example.UserRepository import UserRepository


@MicronautTest
class UserRepositorySpec:

    userRepository: Annotated[UserRepository, Inject]

    @Test
    def testSoftDelete(self):
        fred, bob, joe = self.userRepository.saveAll([
            User("Fred"),
            User("Bob"),
            User("Joe"),
        ])

        self.userRepository.deleteById(joe.id)

        assert self.userRepository.count() == 2
        assert self.userRepository.existsById(fred.id)
        assert not self.userRepository.existsById(joe.id)

        disabled = self.userRepository.findDisabled()
        assert len(disabled) == 1
        assert disabled[0].userName == "Joe"
